samples/CMakeLists.txt

```diff
@@ -8,6 +8,9 @@ add_subdirectory(metadata-read-write)
 add_subdirectory(metadata-manipulation)
 add_subdirectory(std-schema)
 add_subdirectory(metadata-schema)
+add_subdirectory(compression/compression-builtin)
```

Comment:
there is a large intersection between these two samples;
what about joining them?